main.go

```diff
@@ -181,6 +182,10 @@ func main() {
 
 		ul := &StatsDUDPListener{conn: uconn}
 		go ul.Listen(events)
+
+		if runtime.GOOS == "linux" {
```

Comment:
If at all possible, I would prefer that code that doesn't work on non-linux platforms doesn't get compiled in, not just skipped with an `if`.